### ISSUE TITLE
Double click prevention

### ISSUE THREAD

**Issue by laryn** [backdrop/backdrop-issues#6393]
## Description of the need

I was working on a form that has a bit of a lag as it checks something via an external API, and started looking into double click prevention. It turns out it was added to Drupal 7 in 7.98 -- should we add it to Backdrop as well?

 - Change record: https://www.drupal.org/node/3365142
 - Issue: https://www.drupal.org/node/1705618

**Comment by klonos** [comment]
I thought we already sorted that with #2595 🤔 ...there's also #3717 as a follow-up.

**Comment by klonos** [comment]
...right, it seems that we are missing a few things from [this commit](https://git.drupalcode.org/project/drupal/-/commit/353a22fcaecb8310ab01e3ea7100cc2f2a82b315), plus also [this one](https://git.drupalcode.org/project/drupal/-/commit/e523ff5a6dff082ffea2a5604fdde2cc17abe259) which introduces the new `drupal.form-single-submit` library.

So yeah, we should do that 👍🏼 ...but if we brought code parity with these commit, then I believe that we can close #3717 as redundant(?).

**Comment by klonos** [comment]
...I'm assigning this to mysel, to knock it out of the way.

**Comment by klonos** [comment]
So the way things were implemented in D7, the JS that does the double-click prevention was added into a separate file, loaded via the `system_library()`, whereas we have added it to the existing form.js file in Backdrop.

Also, in D7.36 a `javascript_always_use_jquery` setting was introduced ([change record](https://www.drupal.org/node/2462717)), which was before this change here. So the equivalent commit includes tests that take that setting into consideration as well. Plus, the tests are checking if `misc/form-single-submit.js` is loaded on the page when a form exists, but in our case where the code lives in form.js, it will always be loaded, so the tests don't make sense or any point as they were implemented in D7.

So questions:
- do we also introduce `javascript_always_use_jquery` setting or not? (see #3669 and #3671)
- if we do introduce the `javascript_always_use_jquery` setting, do we wait for that before making this change here?
- do we split `Backdrop.behaviors.formSingleSubmit` out of the `form.js` file and load it separately via `system_library()` or leave it as is in `form.js`?

**Comment by herbdool** [comment]
@klonos I don't think we need to wait, and I think it makes sense that the test won't work with the way Backdrop implemented it. So I think the PR is ready for review and testing. The question I have is, what's the best way to manually test to see if double click prevention is working and if it can be disabled by setting `javascript_use_double_submit_protection` to `TRUE`?
